### DISCUSSION TITLE
Creating MacOS app

### DISCUSSION THREAD

**Discussion by talbenarie** [airsdk/Adobe-Runtime-Support#1688] [Q&A]
Packing SWF to MacOS with captive runtime works when i'm using self signed certificate.
However it cannot be installed on other machines / devices [No matter what I tried].
I have tried to generate Mac App Distribution certificate and try to pack captive runtime application however I get the error 
`Unable to build a valid certificate for the signer.`
I am getting very confused and I need help figuring out wat to do to pack proper Mac app and distribute it to others.
I do not necessary want to use the apple app store

**Comment by ajwfrost** [comment]
If you're creating an application on macOS then it should be signed with a "Developer ID Application" certificate (although potentially a "Developer ID Installer" certificate may also work? we just use them for signing .pkg files here though).

The certificate needs to be imported into your KeyChain store. You can either then use this as the 'native' signing options when packaging the bundle, or you can sign it afterwards. See https://github.com/airsdk/Adobe-Runtime-Support/issues/1040#issuecomment-919709012

You'll also need to notarize the app:
https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution?language=objc

If you've created the signed app using ADT then notarization should "just work"...

**Reply by SUSDEV** [reply]
Last year I wrote out detailed instructions that included fixing symbolic links in what adt produced. That was several SDK versions ago, since then, Harman has done a great job to remove the need to fix those links. I just compiled my app with AIR 33.1.1.743 and I went right from creating the app with adt into codesigning and then notarizing and stapling. If you want to see those steps (of course requires what @ajwfrost wrote above first), check out [my April 10 2021 post](https://github.com/airsdk/Adobe-Runtime-Support/discussions/716).
